src/jdk.javadoc/share/classes/jdk/javadoc/internal/doclets/formats/html/taglets/SimpleTaglet.java

```diff
@@ -153,6 +154,27 @@ private boolean isNestedType(Element e) {
                 return e.getKind().isDeclaredType()
                         && ((TypeElement) e).getNestingKind() == NestingKind.MEMBER;
             }
+
+            @Override
+            protected List<? extends BlockTagTree> getBlockTags(Element e) {
+                var tags = utils.getBlockTags(e, a -> a.getKind() == tagKind);
+                if (tags.isEmpty()) {
+                    tags = getDefaultBlockTags(e, this::accepts);
+                }
+                if (tagKind == DocTree.Kind.SINCE
+                        && tags.isEmpty()
+                        && e.getKind().isDeclaredType()
+                        && (e.getEnclosingElement() instanceof PackageElement || isNestedType(e))) {
+                        tags = utils.getBlockTags(getPackageElement(e), a -> a.getKind() == tagKind);
+                }
```

Comment:
Wouldn't this better be done in the `getDefaultBlockTags` method above so we don't have to override `getBlockTags`? 

Also, nitpicking, the static factory method is called `createWithDefaultForNested`, which is not true anymore since also top-level types can now inherit block tags.